### PR TITLE
Security Updates

### PR DESCRIPTION
**Pull request by lucashu1**
* Add CORS whitelist depending on process.env.NODE_ENV
* npm upgrade & audit
* install `helmet`: https://helmetjs.github.io
* (Comment out recommendation-emails cron job)

Some first steps in may what become a series of security-related updates!

Also, I didn't know this till today, but CORS/"origin" is only for browsers (https://stackoverflow.com/questions/36958999/cors-is-it-a-client-side-thing-a-server-side-thing-or-a-transport-level-thin), which is why Insomnia can still make requests to the API without CORS.

### CODE REVIEW

**Comment by lucashu1** [comment]
@nicolengsy @OStouraiti @CodeBubble Tagging you all in case any of you are curious to learn more about CORS: https://www.codecademy.com/articles/what-is-cors

**Comment by Raghav-Maheshwari** [comment]
@lucashu1 I think whitelisting domains is a good start! 
I have been doing some readings (see: https://www.freecodecamp.org/news/never-use-an-absolute-path-for-your-apis-again-9ee9199563be/) that make the case for using a reverse proxy vs whitelisting on cors, but nothing I've read so far is a dealbreaker on using Cors.

File: src/server/util/config.js [hunk 1]
```diff
@@ -1,3 +1,29 @@
+// CORS
+const whitelistProd = [
+  'https://giveduet.org',
+  'https://duet-website-v2.herokuapp.com'
```
Comment (Raghav-Maheshwari) [inline src/server/util/config.js:4]: I think we'd have to whitelist `https://www.giveduet.org`, `http://giveduet.org`, and `http://www.giveduet.org` as well.

**Comment by lucashu1** [comment]
Oo thanks for sharing that link @Raghav-Maheshwari. I'll do more reading and see how difficult it is to get nginx set up before merging this in. It definitely would be nice to have giveduet.org/api/... actually work, especially if the benefits are more than cosmetic.

**Comment by lucashu1** [comment]
Merging this in for now, and will look into re-hosting on nginx to get around CORS. Ideally, giveduet.org/api would point to our API, and giveduet.org/{anything-else} would go to our website.

**Pull request merged**